rubin_nights/consdb_query.py

```diff
@@ -262,15 +315,61 @@ def query(self, query: str) -> pd.DataFrame:
         else:
             messages = response.json()
         if len(messages) > 0:
-            results = pd.DataFrame(messages["data"], columns=messages["columns"])
-            # Check for duplicate columns.
-            indices = np.where(pd.Series(results.columns.duplicated()))[0]
-            newcols = results.columns.to_list()
-            for i in indices:
-                newcols[i] = newcols[i] + "_duplicate"
-            # Have to change only some instances of the duplicates
-            results.columns = newcols
-            results.drop(results.columns[indices], axis=1, inplace=True)
+            results = self._to_pandas(messages)
+        else:
+            results = pd.DataFrame([])
+        return results
+
+    async def async_query(self, query: str) -> pd.DataFrame:
+        """Execute asynchronous FastAPI ConsDB query.
+
+        Parameters
+        ----------
+        query
+            SQL query.
+
+        Returns
+        -------
+        results : `pd.DataFrame`
+        """
+        params = {"query": query}
+        try:
+            response = await self.async_client.post("/query", json=params)
+            if response.status_code == 500:
+                sql_problems = response.json()["message"].replace("\n\n", "\n")
+                if "OperationalError" in sql_problems:
+                    # Just try again - consdb to FastAPI fell asleep?
+                    logger.info(
+                        f"Consdb Operational error at "
+                        f"{datetime.datetime.now(datetime.UTC).strftime('%Y-%m-%d %H:%M:%S')} "
+                        f"- trying again."
+                    )
+                    response = await self.async_client.post("/query", json=params)
+            response.raise_for_status()
+        except httpx.RequestError as exc:
+            logger.error(f"An error occurred while requesting {exc.request.url!r}.")
```

Comment:
You might consider whether it's better to have a single long error message with all information as a unit or multiple log lines that can potentially get interleaved with other messages or otherwise be difficult to correlate.